helix-core/src/syntax.rs

```diff
@@ -106,15 +106,92 @@ pub struct IndentationConfiguration {
     pub unit: String,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum IndentQueryNode {
+    // A node given just by its kind
+    SimpleNode(String),
+    // A node given by a list of characteristics which must all be fulfilled in order to match
+    ComplexNode {
+        kind: Option<String>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Vec::is_empty")]
```

Comment:
Why do we need to skip serializing if empty?

Reply:
I'm actually not sure why the `Configuration` struct needs to be serializable (Helix also compiles without that implementation), so I didn't know how the `Serialize` impl should behave. Let me know what behavior we want here (or if we even need the impl at all).